pandas/core/generic.py

```diff
@@ -10399,37 +10398,323 @@ def _agg_by_level(self, name, axis=0, level=0, skipna=True, **kwargs):
+        numeric_only=None,
+        min_count=0,
+        **kwargs,
+    ):
+        if name == "sum":
+            nv.validate_sum(tuple(), kwargs)
+        elif name == "prod":
+            nv.validate_prod(tuple(), kwargs)
+        else:
+            nv.validate_stat_func(tuple(), kwargs, fname=name)
+        if skipna is None:
+            skipna = True
+        if axis is None:
+            axis = self._stat_axis_number
+        if level is not None:
+            return self._agg_by_level(
+                name, axis=axis, level=level, skipna=skipna, min_count=min_count
+            )
+        return self._reduce(
+            func,
+            name=name,
+            axis=axis,
+            skipna=skipna,
+            numeric_only=numeric_only,
+            min_count=min_count,
+        )
+
+    def sum(
+        self,
+        axis=None,
+        skipna=None,
+        level=None,
+        numeric_only=None,
+        min_count=0,
+        **kwargs,
+    ):
+        return self._min_count_stat_function(
+            "sum", nanops.nansum, axis, skipna, level, numeric_only, min_count, **kwargs
+        )
+
+    def prod(
+        self,
+        axis=None,
+        skipna=None,
+        level=None,
+        numeric_only=None,
+        min_count=0,
+        **kwargs,
+    ):
+        return self._min_count_stat_function(
+            "prod",
+            nanops.nanprod,
+            axis,
+            skipna,
+            level,
+            numeric_only,
+            min_count,
+            **kwargs,
+        )
+
+    product = prod
+
+    def mad(self, axis=None, skipna=None, level=None):
+        """
+        {desc}
+
+        Parameters
+        ----------
+        axis : {axis_descr}
+            Axis for the function to be applied on.
+        skipna : bool, default None
+            Exclude NA/null values when computing the result.
+        level : int or level name, default None
+            If the axis is a MultiIndex (hierarchical), count along a
+            particular level, collapsing into a {name1}.
+
+        Returns
+        -------
+        {name1} or {name2} (if level specified)\
+        {see_also}\
+        {examples}
+        """
+        if skipna is None:
+            skipna = True
+        if axis is None:
+            axis = self._stat_axis_number
+        if level is not None:
+            return self._agg_by_level("mad", axis=axis, level=level, skipna=skipna)
+
+        data = self._get_numeric_data()
+        if axis == 0:
+            demeaned = data - data.mean(axis=0)
+        else:
+            demeaned = data.sub(data.mean(axis=1), axis=0)
+        return np.abs(demeaned).mean(axis=axis, skipna=skipna)
+
     @classmethod
     def _add_numeric_operations(cls):
         """
         Add the operations to the cls; evaluate the doc strings again
         """
         axis_descr, name1, name2 = _doc_parms(cls)
 
-        cls.any = _make_logical_function(
-            cls,
-            "any",
+        @doc(
```

Comment:
is the point of leaving these here, to make the doc strings work / make mypy happy? e.g. you defined them above an now you are re-defining them for Series/DataFrame

Reply:
> it'd be nice to have a less-verbose way of pinning the docstrings. i haven't had any luck so far at making that work.

i guess this comment explains?

Reply:
redefining to get the docstrings right.  im not wild about this either

Reply:
yah.  im going to try @simonjayhawkins's suggestion, so let's stick a pin in this, ill gentle-ping when ready

Reply:
that was my initial reaction. but am now more comfortable that this could be dealt with in a follow-on

Reply:
either way, lgtm.